azuredevops/internal/service/serviceendpoint/data_serviceendpoint_jfrog_artifactory_v2.go

```diff
@@ -0,0 +1,100 @@
+package serviceendpoint
+
+import (
+	"fmt"
+	"maps"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/validate"
+)
+
+// DataSourceServiceEndpointJFrogArtifactoryV2 schema and implementation for JFrog Artifactory service endpoint resource
+func DataSourceServiceEndpointJFrogArtifactoryV2() *schema.Resource {
+	r := &schema.Resource{
+		Read: DataSourceServiceEndpointJFrogArtifactoryV2Read,
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(1 * time.Minute),
+		},
+		Importer: tfhelper.ImportProjectQualifiedResourceUUID(),
+		Schema:   baseSchema(),
+	}
+
+	maps.Copy(r.Schema, map[string]*schema.Schema{
+		"url": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validate.Url,
+			Description:  "Url for the JFrog Artifactory Server",
+		},
+
+		"authentication_token": {
```

Comment:
Service  does not return sensitive properties, can remove them in the data source.